endpointmanager/pkg/capabilityhandler/capabilityreceiver.go

```diff
@@ -1 +1,124 @@
 package capabilityhandler
+
+import (
+	"context"
+	"database/sql"
+	"encoding/json"
+	"strings"
+	"time"
+
+	"github.com/onc-healthit/lantern-back-end/capabilityquerier/pkg/capabilityquerier"
+
+	"github.com/spf13/viper"
+
+	"github.com/onc-healthit/lantern-back-end/capabilityquerier/pkg/queue"
+	"github.com/onc-healthit/lantern-back-end/endpointmanager/pkg/capabilityhandler/config"
+
+	"github.com/onc-healthit/lantern-back-end/endpointmanager/pkg/endpointmanager"
+	log "github.com/sirupsen/logrus"
+)
+
+func failOnError(err error) {
+	if err != nil {
+		log.Fatalf("%s", err)
+	}
+}
+
+func formatMessage(message []byte) (*endpointmanager.FHIREndpoint, error) {
+	var msgJSON capabilityquerier.Message
+
+	err := json.Unmarshal(message, &(msgJSON))
+	if err != nil {
+		return nil, err
+	}
+
+	originalURL := strings.Replace(msgJSON.URL, "metadata", "", 1)
+
+	fhirEndpoint := endpointmanager.FHIREndpoint{
+		URL:                 originalURL,
+		TLSVersion:          msgJSON.TLSVersion,
+		MimeType:            msgJSON.MimeType,
+		CapabilityStatement: msgJSON.CapabilityStatement,
```

Comment:
recommend also capturing the errors that were observed (msg.JSON.Errs, or whatever it is)

Reply:
should the errors just be logged in this function? or should they be returned and prevent the capability statement & metadata from being added to the database? or should they be saved in the database as more metadata information?

Reply:
saved as metadata

Reply:
👍